### PR TITLE
fix: Clear errors after validation on submit

### PR DESCRIPTION
**Pull request by taion**

### CODE REVIEW

**Comment by jquense** [comment]
how hard would it be to write a test for this? Want to make sure this doesn't break quietly in the future

File: src/Form.tsx [hunk 1]
```diff
@@ -386,6 +386,7 @@ const _Form: Form = React.forwardRef(
     };
 
     const handleSubmitSuccess = (validatedValue: InferType<T>) => {
+      notify(onError, []);
```
Comment (jquense) [inline src/Form.tsx:389]: maybe only if there are errors? or maybe it should be more finely grained and only clear the errors that were queued when submit started? IDK, clearing errors on a submit seems like reasonable behavior so probably over thinking it

Comment (taion) [reply src/Form.tsx:389]: i'm paralleling the logic from https://github.com/taion/react-formal/blob/a68c2b13c77dbdac2ba2e0daeb691291f0d63fb6/src/Form.tsx#L420 , which overwrites all the errors with the validation errors from https://github.com/taion/react-formal/blob/a68c2b13c77dbdac2ba2e0daeb691291f0d63fb6/src/Form.tsx#L456-L460

**Comment by taion** [comment]
@jquense i can look at tests, but can you confirm that this approach is okay with you?

**Comment by jquense** [comment]
ya LGTM

**Comment by taion** [comment]
@jquense i added a quick test. it's very unit-y and it's specifically focused on just this one state, but i think it should prevent regressions? wdyt

**Comment by taion** [comment]
༼ つ ◕_◕ ༽つ Give v2.1.2

**Comment by jquense** [comment]
done!

**Pull request merged**